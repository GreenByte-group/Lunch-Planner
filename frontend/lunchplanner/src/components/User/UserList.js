import React from 'react';
import {withStyles} from 'material-ui/styles';
import {List} from "material-ui";
import User from "./User";

const styles = {
    list: {
        padding: 0,
        marginTop: '0px !important',
    },
};

class UserList extends React.Component {

    constructor(props) {
        super();

        this.state = {
            selectedUsers: props.selectedUsers || [],
            users: props.users || [],
            selectable: props.selectable || false,
            onSelectionChanged: props.onSelectionChanged,
            othersInvited: props.othersInvited || false,
        };
    }

    componentWillReceiveProps(newProps) {
        let selectedUsers, users;

        if(newProps.selectedUsers && newProps.selectedUsers !== this.state.selectedUsers) {
            selectedUsers = newProps.selectedUsers;
            this.setState({
                selectedUsers: selectedUsers,
            });
        }

        if(newProps.users && newProps.users !== this.state.users) {
            users = newProps.users;
            this.setState({
                users: users,
            });
        }
    }

    clickHandler = (username, selected) => {
        if(this.state.selectable) {
            let selectedUsers = this.state.selectedUsers;
            if (selected) {
                selectedUsers.push(username);
            } else {
                let index = selectedUsers.indexOf(username);
                selectedUsers.splice(index, 1);
            }

            this.setState({
                selectedUsers: selectedUsers,
            });

            this.state.onSelectionChanged(selectedUsers);
        }
    };

    render() {
        const { classes } = this.props;
        let users = this.state.users;
        let selectedUsers = this.state.selectedUsers;
        let othersInvited = this.state.othersInvited;

        return (
            <List
                className={classes.list}
            >
                {users.map((listValue) => {
                    return <User invited={othersInvited} selectable={this.state.selectable} selected={selectedUsers.includes(listValue.userName)} username={listValue.userName} onClick={this.clickHandler}/>;
                })}
            </List>
        );
    }
}

export default withStyles(styles, { withTheme: true })(UserList);