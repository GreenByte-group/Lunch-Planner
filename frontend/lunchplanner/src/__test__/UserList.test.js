import React from 'react';
import UserList from '../components/User/UserList';
import renderer from 'react-test-renderer';
import {MemoryRouter} from "react-router-dom";

it ("renders correctly", ()=>{
    const tree = renderer.create(
        <UserList/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});