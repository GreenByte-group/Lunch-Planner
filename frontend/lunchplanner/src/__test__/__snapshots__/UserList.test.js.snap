// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="MuiList-root-2 MuiList-padding-3 UserList-list-1"
/>
`;
